File: LoginApplication/src/components/CustomerHome/Get/Coupon.js
import React from 'react';

import classes from './Coupon.module.css';

const Coupon = (props) => {
  return (
    <li className={classes.Coupon}>
      <h2>{props.category}</h2>
      <p>{props.id}</p>
      <p>{props.company}</p>
      <h3>{props.title}</h3>
      <p>{props.description}</p>
      <h5>{props.amount}</h5>
      <h5>{props.startDate}</h5>
      <h5>{props.endDate}</h5>
      <h5>{props.price}</h5>
      <img src={props.image} alt="" />

    </li>
  );
};

export default Coupon;

File: cupon_server/src/main/java/com/evgenie_tomer_itay/entities/Coupon.java
package com.evgenie_tomer_itay.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString.Exclude;

import javax.persistence.*;
import java.time.LocalDate;
import java.util.List;

@Entity
@Data
@Table(name = "coupons")
@AllArgsConstructor
@NoArgsConstructor
@Builder

public class Coupon {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int id;

	@ManyToOne(fetch = FetchType.EAGER)
	@JoinColumn(name = "company_id")
	private Company company;

	@ManyToMany(fetch = FetchType.LAZY)
	@JoinTable(name = "customers_vs_coupons", joinColumns = @JoinColumn(name = "coupon_id"), inverseJoinColumns = @JoinColumn(name = "customer_id"))
	@Exclude
	@JsonIgnore
	private List<Customer> customers;

	@Enumerated(EnumType.STRING)
	@Column(nullable = false)
	private Category category;
	
	@Column(nullable = false)
	private String title;
	
	@Column(nullable = false)
	private String description;
	
	@Column(nullable = false)
	private LocalDate startDate;
	
	@Column(nullable = false)
	private LocalDate endDate;
	
	@Column(nullable = false)
	private Integer amount;
	
	@Column(nullable = false)
	private Double price;
	
	@Column(nullable = false)
	private String image;

}
